src/matrix/operator.cpp

```diff
@@ -356,6 +356,18 @@ struct MatrixHomoToPoseQuaternion
 };
 REGISTER_UNARY_OP(MatrixHomoToPoseQuaternion, MatrixHomoToPoseQuaternion);
 
+struct PoseQuaternionToMatrixHomo
+    : public UnaryOpHeader<Vector, MatrixHomogeneous> {
+  void operator()(const dg::Vector &vect, MatrixHomogeneous &Mres) {
+    VectorQuaternion quat;
+    quat.w() = vect[6];
+    quat.vec() = vect.segment<3>(3);
+    Mres.translation() = vect.head<3>();
+    Mres.linear() = quat.toRotationMatrix();
```

Comment:
Minor comment: I prefer the shorter version below.
```suggestion
    Mres.translation() = vect.head<3>();
    Mres.linear() = VectorQuaternion(vect.tail<4>());
```

Reply:
Yes I totally agree !

Reply:
Just a comment, I think for the linear part it is:
`Mres.linear() = VectorQuaternion(vect.tail<4>()).toRotationMatrix();`

Reply:
There may be a `toRotationMatrix` missing.